### ISSUE TITLE
OptionManagement.validateOption throws NoSuchEntityException for "0" option label

### ISSUE THREAD

**Issue by rzinnatullin** [magento/magento2#13083]
### Preconditions

1. Magento 2.1.11 (and 2.2.2 I suppose)

### Steps to reproduce
1. Create a product attribute of 'dropdown' type via admin page with one option (Admin value: 0, Default Store View value: 0)
2. Programmatically delete the option using OptionManagement.delete method
```PHP
    /**
     * @var \Magento\Eav\Api\AttributeOptionManagementInterface
     */
    private $optionManagement;
...
    $this->optionManagement->delete(\Magento\Catalog\Model\Product::ENTITY, $attributeCode, $optionId);
```

### Expected result
1. The option should be delete w/o any errors

### Actual result
1. `NoSuchEntityException` with message `Attribute <somecode> does not contain option with Id <someId>` is thrown

### Root cause
There is `validateOption` method in `OptionManagement.php` that checks if option exists:
![image](https://user-images.githubusercontent.com/11244866/34757861-cb8e4c16-f5f5-11e7-9e3e-e76a8a85cdc9.png)

`getOptionText` returns "0" string as it is supposed to:
![image](https://user-images.githubusercontent.com/11244866/34757883-eb845f9c-f5f5-11e7-86b0-a013be9e6026.png)

So the root cause is in `validateOption` method (see the first screenshot). In line 135 it does logical NOT operator, so "0" is casted to FALSE, thus the condition is TRUE. To fix the issue the comparison should be exact: `=== FALSE`.

**Comment by orlangur** [comment]
Hi @rzinnatullin, thanks for comprehensive investigation!

Are you interested in preparing a pull request maybe? They are processed pretty fast nowadays, it would speed up fix availability for everyone and also it is easier to discuss concrete code snippet.

**Comment by magento-engcom-team** [comment]
@rzinnatullin, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by sidolov** [comment]
Hi @rzinnatullin. Thank you for your report. 
The issue has been fixed in magento/magento2#18720 by @pmclain in 2.3-develop branch
Related commit(s):
 - [ed6a2fba850caa5e2402663fee133d4f56545200](https://github.com/magento/magento2/commit/ed6a2fba850caa5e2402663fee133d4f56545200)

The fix will be available with the upcoming 2.3.1 release.

**Comment by magento-engcom-team** [comment]
Hi @rzinnatullin. Thank you for your report. 
The issue has been fixed in magento/magento2#18873 by @gelanivishal in 2.2-develop branch
Related commit(s):
 - [06ca13a554082ed8fcbe7852d765aa2916b48fb1](https://github.com/magento/magento2/commit/06ca13a554082ed8fcbe7852d765aa2916b48fb1)

The fix will be available with the upcoming 2.2.8 release.
